src/main/java/umaikaze/duke/Duke.java

```diff
@@ -0,0 +1,74 @@
+package umaikaze.duke;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.time.format.DateTimeParseException;
+
+public class Duke {
+    Ui ui;
+    Storage st;
+    TaskList tl;
+
+    public Duke(String savePath) throws UnsupportedEncodingException {
+        ui = new Ui();
+        st = new Storage(savePath);
+        try {
+            tl = new TaskList(st.loadFile());
+        } catch (IOException e) {
+            ui.showError(e.getMessage());
+            tl = new TaskList();
+        }
+    }
+
+    void run(BufferedReader br) throws IOException {
+        ui.showReply("how may i sewve u today nya?");
+        String[] line = br.readLine().split(" ");
+        String cmd = line[0];
+        while (!cmd.equals("bye")) {
```

Comment:
`ui.class` can be used to process user input, categorize them, and display things to back to the user.
You might consider put this block into `ui.class` to simplify you main class.
:D